File: backend/app/modules/console/api/doc_center/design_module.py
"""
设计对接模块接口
"""

from typing import Optional

from fastapi import APIRouter, Depends, Query
from sqlalchemy.ext.asyncio import AsyncSession

from app.common.response import fail, success
from app.core.dependencies import get_current_user, get_platform_db
from app.core.security import TokenData
from app.modules.console.schemas.doc_center.design_module import (
    DesignModuleCreate,
    DesignModuleOut,
    DesignModulePriorityUpdate,
    DesignModuleSortRequest,
    DesignModuleStatusUpdate,
    DesignModuleUpdate,
)
from app.modules.console.services.doc_center.design_module_service import (
    DesignModuleService,
)

router = APIRouter()


def _to_out(row) -> dict:
    return DesignModuleOut.model_validate(row).model_dump(mode="json")


@router.get("")
async def list_design_modules(
    page: int = Query(1, ge=1),
    limit: int = Query(20, ge=1, le=200),
    status: Optional[int] = Query(None, description="状态筛选"),
    priority: Optional[int] = Query(None, description="优先级筛选"),
    product_line: Optional[str] = Query(None, description="产品端筛选"),
    keyword: Optional[str] = Query(None, description="关键词"),
    view: Optional[str] = Query(None, description="list|board"),
    db: AsyncSession = Depends(get_platform_db),
    _: TokenData = Depends(get_current_user),
):
    """设计对接模块列表 / 看板"""
    if view == "board":
        board = await DesignModuleService.list_board(
            db,
            priority=priority,
            product_line=product_line,
            keyword=keyword,
        )
        return success(
            data={
                "board": {
                    k: [_to_out(i) for i in v] for k, v in board.items()
                }
            }
        )

    items, total = await DesignModuleService.list_page(
        db,
        page=page,
        limit=limit,
        status=status,
        priority=priority,
        product_line=product_line,
        keyword=keyword,
    )
    return success(
        data={
            "list": [_to_out(i) for i in items],
            "total": total,
            "page": page,
            "limit": limit,
        }
    )


@router.put("/sort")
async def sort_design_modules(
    data: DesignModuleSortRequest,
    db: AsyncSession = Depends(get_platform_db),
    current_user: TokenData = Depends(get_current_user),
):
    """批量更新排序（支持跨列改状态）"""
    await DesignModuleService.sort(db, data.items, current_user.user_id)
    return success(message="排序已更新")


@router.post("")
async def create_design_module(
    data: DesignModuleCreate,
    db: AsyncSession = Depends(get_platform_db),
    current_user: TokenData = Depends(get_current_user),
):
    """创建设计对接模块"""
    row = await DesignModuleService.create(db, data, current_user.user_id)
    return success(data=_to_out(row), message="已创建模块")


@router.get("/{module_id}")
async def get_design_module(
    module_id: int,
    db: AsyncSession = Depends(get_platform_db),
    _: TokenData = Depends(get_current_user),
):
    """设计对接模块详情"""
    row = await DesignModuleService.get_by_id(db, module_id)
    if not row:
        return fail("未找到该模块，请刷新后重试")
    return success(data=_to_out(row))


@router.put("/{module_id}")
async def update_design_module(
    module_id: int,
    data: DesignModuleUpdate,
    db: AsyncSession = Depends(get_platform_db),
    current_user: TokenData = Depends(get_current_user),
):
    """更新设计对接模块"""
    row = await DesignModuleService.update(
        db, module_id, data, current_user.user_id
    )
    return success(data=_to_out(row), message="已保存")


@router.patch("/{module_id}/status")
async def update_design_module_status(
    module_id: int,
    data: DesignModuleStatusUpdate,
    db: AsyncSession = Depends(get_platform_db),
    current_user: TokenData = Depends(get_current_user),
):
    """更新状态"""
    row = await DesignModuleService.update_status(
        db, module_id, data.status, current_user.user_id
    )
    return success(data=_to_out(row), message="状态已更新")


@router.patch("/{module_id}/priority")
async def update_design_module_priority(
    module_id: int,
    data: DesignModulePriorityUpdate,
    db: AsyncSession = Depends(get_platform_db),
    current_user: TokenData = Depends(get_current_user),
):
    """更新优先级"""
    row = await DesignModuleService.update_priority(
        db, module_id, data.priority, current_user.user_id
    )
    return success(data=_to_out(row), message="优先级已更新")


@router.delete("/{module_id}")
async def delete_design_module(
    module_id: int,
    db: AsyncSession = Depends(get_platform_db),
    current_user: TokenData = Depends(get_current_user),
):
    """删除设计对接模块（软删）"""
    await DesignModuleService.delete(db, module_id, current_user.user_id)
    return success(message="已删除该模块")

File: frontend/console/src/api/doc-center/design-module.ts
import request from '@/utils/request';
import type { ApiResult } from '@/api';
import type {
  DesignModule,
  DesignModuleForm,
  DesignModuleParam,
  DesignModuleSortItem
} from './model/design-module';

/**
 * 分页列表
 */
export async function pageDesignModules(params: DesignModuleParam) {
  const res = await request.get<
    ApiResult<{
      list: DesignModule[];
      total: number;
      page: number;
      limit: number;
    }>
  >('/doc-center/design-modules', { params });
  if (res.data.code === 0) {
    return res.data.data;
  }
  return Promise.reject(new Error(res.data.message));
}

/**
 * 看板数据（按状态分组）
 */
export async function boardDesignModules(params?: DesignModuleParam) {
  const res = await request.get<
    ApiResult<{ board: Record<string, DesignModule[]> }>
  >('/doc-center/design-modules', {
    params: { ...params, view: 'board' }
  });
  if (res.data.code === 0) {
    return res.data.data?.board ?? {};
  }
  return Promise.reject(new Error(res.data.message));
}

/**
 * 详情
 */
export async function getDesignModule(id: number) {
  const res = await request.get<ApiResult<DesignModule>>(
    `/doc-center/design-modules/${id}`
  );
  if (res.data.code === 0) {
    return res.data.data;
  }
  return Promise.reject(new Error(res.data.message));
}

/**
 * 创建
 */
export async function createDesignModule(data: DesignModuleForm) {
  const res = await request.post<ApiResult<DesignModule>>(
    '/doc-center/design-modules',
    data
  );
  if (res.data.code === 0) {
    return res.data.data;
  }
  return Promise.reject(new Error(res.data.message));
}

/**
 * 更新
 */
export async function updateDesignModule(id: number, data: DesignModuleForm) {
  const res = await request.put<ApiResult<DesignModule>>(
    `/doc-center/design-modules/${id}`,
    data
  );
  if (res.data.code === 0) {
    return res.data.data;
  }
  return Promise.reject(new Error(res.data.message));
}

/**
 * 更新状态
 */
export async function updateDesignModuleStatus(id: number, status: number) {
  const res = await request.patch<ApiResult<DesignModule>>(
    `/doc-center/design-modules/${id}/status`,
    { status }
  );
  if (res.data.code === 0) {
    return res.data.data;
  }
  return Promise.reject(new Error(res.data.message));
}

/**
 * 更新优先级
 */
export async function updateDesignModulePriority(id: number, priority: number) {
  const res = await request.patch<ApiResult<DesignModule>>(
    `/doc-center/design-modules/${id}/priority`,
    { priority }
  );
  if (res.data.code === 0) {
    return res.data.data;
  }
  return Promise.reject(new Error(res.data.message));
}

/**
 * 批量排序
 */
export async function sortDesignModules(items: DesignModuleSortItem[]) {
  const res = await request.put<ApiResult<unknown>>(
    '/doc-center/design-modules/sort',
    { items }
  );
  if (res.data.code === 0) {
    return res.data.message;
  }
  return Promise.reject(new Error(res.data.message));
}

/**
 * 删除
 */
export async function removeDesignModule(id: number) {
  const res = await request.delete<ApiResult<unknown>>(
    `/doc-center/design-modules/${id}`
  );
  if (res.data.code === 0) {
    return res.data.message;
  }
  return Promise.reject(new Error(res.data.message));
}
